src/primitives/transaction.h

```diff
@@ -304,6 +314,16 @@ struct CMutableTransaction
     uint256 GetHash() const;
 
     std::string ToString() const;
+
+    bool IsSpecialTxVersion() const
```

Comment:
let's drop it from CMutableTransaction which is **_struct_** not **_class_**.
Let's keep it simple as possible.

Check [this,](https://github.com/knst/dash/commit/986ce23fd9569d205aaabb017250243724f80cd0) [this](https://github.com/knst/dash/commit/0d3758a74af2ff28a0efd076ae69a871cbb3db3b) and [finally this](https://github.com/knst/dash/commit/f9c04d62d806fea5b0035010f6858d54da937d6c)

Reply:
It's ok-ish to have small simple helpers in struct imo. Bitcoin even has a slightly complicated one https://github.com/bitcoin/bitcoin/blob/master/src/primitives/transaction.h#L412-L420. I also like 2 initial patches you provided btw, pls submit them in a separate PR if we agree on having helpers here.

Reply:
but these helpers in CMutableTransaction are not actually useful, are they?

Reply:
that's true for `IsSpecialTxVersion` probably cause it's used once and is used internally only but `HasExtraPayloadField` avoids hardcoding tx version in `src/script/interpreter.cpp` which looks valuable

Reply:
`src/script/interpreter.cpp` use low-level own implementtion of serialization.

just look to the couple lines above:
```
        int32_t n32bitVersion = txTo.nVersion | (txTo.nType << 16);
```

it knows too much, that's still internal usage IMO even if it is outside of struct

Reply:
Hmm, true... ok, you convinced me I guess :D

Reply:
makes sense, one small suggestion https://github.com/knst/dash/commit/b645a1d3d34bec08e16f576ae03eb871d22d4e50#r138441182